lstchain/reco/dl1_to_dl2.py

```diff
@@ -622,7 +622,11 @@ def apply_models(dl1,
                                             + config['particle_classification_features']
                                             + config['disp_classification_features'],
                               )
-
+    # if no events in dl2, e.g. for bad time interval from Cat-B calibration
+    if len(dl2) == 0:
+        logger.warning("No events in dl2. Return.")
+        return dl2
```

Comment:
I think this should exit and not write a file.

Reply:
mmh, not sure, you will have missing subruns in the list of files, is this ok?

Reply:
I don't think that is a problem. Not completely sure, though

Reply:
me too :-)  

@maxnoe, @moralejo, @rlopezcoto do you have an opinion on that?

Reply:
Is this an issue for the live time computation?

Reply:
I just realised the empty dl2 files gives problems to the merge tool: 
```
Can't append node /dl2/event/telescope/parameters/LST_LSTCam from file ./dl2_LST-1.Run12051.0016.h5
Traceback (most recent call last):
  File "/fefs/aswg/workspace/franca.cassol/software/miniconda3/envs/lst-ctapipe0.17/bin/lstchain_merge_hdf5_files", line 8, in <module>
    sys.exit(main())
  File "/fefs/aswg/workspace/franca.cassol/software/cta-lstchain/lstchain/scripts/lstchain_merge_hdf5_files.py", line 94, in main
    auto_merge_h5files(
  File "/fefs/aswg/workspace/franca.cassol/software/cta-lstchain/lstchain/io/io.py", line 355, in auto_merge_h5files
    out_node.append(in_node.read().astype(out_node.dtype))
TypeError: Cannot cast array data from dtype([('obs_id', '<i8'), ('event_id', '<i8'), ('intensity', '<f8'), ('log_intensity', '<f8'), ('x', '<f8'), ('y', '<f8'), ('r', '<f8'), ('phi', '<f8'), ('length', '<f8'), ('length_uncertainty', '<f8'), ('width', '<f8'), ('width_uncertainty', '<f8'), ('psi', '<f8'), ('skewness', '<f8'), ('kurtosis', '<f8'), ('time_gradient', '<f8'), ('intercept', '<f8'), ('leakage_intensity_width_1', '<f4'), ('leakage_intensity_width_2', '<f4'), ('leakage_pixels_width_1', '<f8'), ('leakage_pixels_width_2', '<f8'), ('n_pixels', '<i8'), ('concentration_cog', '<f8'), ('concentration_core', '<f8'), ('concentration_pixel', '<f8'), ('n_islands', '<i8'), ('alt_tel', '<f8'), ('az_tel', '<f8'), ('sin_az_tel', '<f8'), ('calibration_id', '<i8'), ('dragon_time', '<f8'), ('ucts_time', '<i8'), ('tib_time', '<i8'), ('ucts_jump', '?'), ('mc_type', '<i8'), ('wl', '<f8'), ('tel_id', '<i8'), ('tel_pos_x', '<f8'), ('tel_pos_y', '<f8'), ('tel_pos_z', '<f8'), ('trigger_type', 'u1'), ('ucts_trigger_type', 'u1'), ('trigger_time', '<f8'), ('event_type', '<i8'), ('delta_t', '<f8')]) to dtype([('obs_id', '<i8'), ('event_id', '<i8'), ('intensity', '<f8'), ('log_intensity', '<f8'), ('x', '<f8'), ('y', '<f8'), ('r', '<f8'), ('phi', '<f8'), ('length', '<f8'), ('length_uncertainty', '<f8'), ('width', '<f8'), ('width_uncertainty', '<f8'), ('psi', '<f8'), ('skewness', '<f8'), ('kurtosis', '<f8'), ('time_gradient', '<f8'), ('intercept', '<f8'), ('leakage_intensity_width_1', '<f4'), ('leakage_intensity_width_2', '<f4'), ('leakage_pixels_width_1', '<f8'), ('leakage_pixels_width_2', '<f8'), ('n_pixels', '<i8'), ('concentration_cog', '<f8'), ('concentration_core', '<f8'), ('concentration_pixel', '<f8'), ('n_islands', '<i8'), ('alt_tel', '<f8'), ('az_tel', '<f8'), ('sin_az_tel', '<f8'), ('calibration_id', '<i8'), ('dragon_time', '<f8'), ('ucts_time', '<i8'), ('tib_time', '<i8'), ('ucts_jump', '?'), ('mc_type', '<i8'), ('wl', '<f8'), ('tel_id', '<i8'), ('tel_pos_x', '<f8'), ('tel_pos_y', '<f8'), ('tel_pos_z', '<f8'), ('trigger_type', 'u1'), ('ucts_trigger_type', 'u1'), ('trigger_time', '<f8'), ('event_type', '<i8'), ('delta_t', '<f8'), ('log_reco_energy', '<f8'), ('reco_energy', '<f8'), ('reco_disp_norm', '<f8'), ('reco_disp_sign', '<i8'), ('reco_disp_sign_proba', '<f8'), ('reco_disp_dx', '<f8'), ('reco_disp_dy', '<f8'), ('reco_src_x', '<f8'), ('reco_src_y', '<f8'), ('signed_time_gradient', '<f8'), ('signed_skewness', '<f8'), ('reco_alt', '<f8'), ('reco_az', '<f8'), ('reco_type', '<i8'), ('gammaness', '<f8')]) according to the rule 'unsafe'
```

If there are not arguments against it, do I change this in order not to write the dl2 file if empty?

Reply:
@max, @morcuended,  after discussion in the  f2f meeting, @moralejo suggested not to remove the events in the bad time intervals but to add a boolean tag, e.g. GTI, which should permit to mask events from bad  periods and remove the period from the lifetime, but keeping the events in the dl2. This would avoid also to create empty dl2. What do you think?

Reply:
I agree, removing events is bad I think.

Reply:
On the other side, this means to add a new column to the DL1ParametersContainer, which we want to move toward ctapipe standards. Since this part must be still decided in ctapipe calibration  concept (and data model),  I can perhaps, for the moment, just create a filter function, based on the dl1 catB monitoring data, to be runned before dealing with a dl2 dataframes analysis ...